src/cls/rbd/cls_rbd.cc

```diff
@@ -2649,13 +2649,17 @@ int metadata_list(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
     return -EINVAL;
   }
 
+  if (max_return == 0) {
+    max_return = RBD_MAX_KEYS_READ;
```

Comment:
Then we would need to update at least the comment above ("0 means no limit")? And it seems in this case the client has a no way to be sure all values are returned until it try to read more.

It looks like it is the only "list" function here that has a special meaning for max_return = 0. May we remove this, so 0 just means 0?

Reply:
I have a separate ticket [1] to update the rbd CLI to iterate which will be backported that I can include w/ this PR. I wouldn't want to change "max_return == zero" to be the more correct actually return zero results until we don't have to worry about older client versions anymore.

[1] http://tracker.ceph.com/issues/21179